### PR TITLE
Owned objects API for service principal

### PR DESCRIPTION
**Pull request by tsologub**
Add support for service principal "owned objects" API. Hidden, but quite handy functionality.

### CODE REVIEW

File: msgraph/serviceprincipals_test.go [hunk 1]
```diff
@@ -157,6 +176,70 @@ func testServicePrincipalsClient_ListGroupMemberships(t *testing.T, c ServicePri
...
+		t.Fatalf("ServicePrincipalsClient.ListOwners(): %v", err)
+	}
+
+	if status < 200 || status >= 300 {
+		t.Fatalf("ServicePrincipalsClient.ListOwners(): invalid status: %d", status)
+	}
+
+	if len(*owners) != 1 {
+		t.Fatalf("ServicePrincipalsClient.ListOwners(): expected owners length 1. was: %d", len(*owners))
+	}
```
Comment (manicminer) [inline msgraph/serviceprincipals_test.go:220]: I'm getting some failures here as it seems the SP inherits the application owners, which, in my testing tenant, gets autoassigned the calling SP as owner at creation time. This works around it for me, WDYT?

```suggestion
	ownersExpected := len(expected)

	if len(*owners) < ownersExpected {
		t.Fatalf("ServicePrincipalsClient.ListOwners(): expected at least 1 owner. has: %d", len(*owners))
	}

	var ownersFound int
	for _, e := range expected {
		for _, o := range *owners {
			if e == o {
				ownersFound++
				continue
			}
		}
	}
	if ownersFound < ownersExpected {
		t.Fatalf("ServicePrincipalsClient.ListOwners(): expected %d matching owners. found: %d", ownersExpected, ownersFound)
	}
```

**Review by manicminer — COMMENTED** [review]
Thanks for this PR @tsologub! This mostly LGTM, I'm just getting some test failures for which I've proposed a workaround. Let me know what you think and once we have a solid test this should be good to merge :)

File: msgraph/serviceprincipals_test.go (hunk 1 above)
Comment (tsologub) [inline msgraph/serviceprincipals_test.go:220]:
```suggestion
	if len(*owners) == 0 {
		t.Fatalf("ServicePrincipalsClient.ListOwners(): expected at least 1 owner. has: %d", len(*owners))
	}
```
As I don't know the exact number of `expected` owners(your suggestion contains a non-existing `expected` variable), what would you say if we check as follows?

Comment (manicminer) [reply msgraph/serviceprincipals_test.go:220]: My bad! I missed that when pasting. `expected` is an extra function arg - a string slice containing any expected owner UUIDs. The goal being to ensure at least a specific list of owners is correctly assigned.

**Review by manicminer — APPROVED** [review]
Sorry for the delay in re-reviewing @tsologub, this looks great and passes for me. Thanks again!

**Comment by tsologub** [comment]
@manicminer no problem. When are you planning to make a `0.13.0` release?

**Comment by manicminer** [comment]
@tsologub Should be sometime in the next day or so

**Pull request merged**